doc/tutorials/enable_s5.rst

```diff
@@ -0,0 +1,367 @@
+
+* For RT-Linux, refer to the ``devicemodel/samples/nuc/launch_hard_rt_vm.sh`` script:
+
+  .. code-block:: none
+     :emphasize-lines: 2,3,12
+     :name: rt-script
+     :caption: RT-Linux launch script
+  
+     # for pm setting
+     pm_channel="--pm_notify_channel uart "
+     pm_by_vuart="--pm_by_vuart tty,/dev/ttyS1"
+     
+     /usr/bin/acrn-dm -A -m $mem_size -s 0:0,hostbridge \
+        --lapic_pt \
+        --rtvm \
+        --virtio_poll 1000000 \
+        -U 495ae2e5-2603-4d64-af76-d4bc5a8ec0e5 \
+        -s 2,passthru,02/0/0 \
+        -s 3,virtio-console,@stdio:stdio_port \
+        $pm_channel $pm_by_vuart \
+        --ovmf /usr/share/acrn/bios/OVMF.fd \
+        hard_rtvm
+     
+     }
+
+  .. note:: For RT-Linux, the vUART is emulated in hypervisor, expose the node as ``/dev/ttySn``.
+
+#. For LaaG and RT-Linux VM, a reference daemon is offered to run:
+
+   a. Use below command lines to make ``life_mngr``.
+
+      .. code-block:: none
+   
+         $ cd acrn-hypervisor/misc/life_mngr
+         $ make life_mngr
+
+   #. Copy ``life_mngr`` and ``life_mngr.service`` into User VM:
+      ``/usr/bin/life_mngr``, and ``/lib/systemd/system/life_mngr.service``
+
+   #. Use below commands to enable ``life_mngr.service`` and restart User VM.
+
+      .. code-block:: none
+
+         # chmod +x /usr/bin/life_mngr
+         # systemctl enable life_mngr.service
+         # reboot
+
+#. For WaaG VM, a reference daemon is offered to run:
+
+   a) Build ``life_mngr_win.exe`` application::
+
+        # cd acrn-hypervisor/misc
+        # make life_mngr
+
+      .. note:: If there is no ``x86_64-w64-mingw32-gcc`` compiler, you need run ``swupd bundle-add c-basic-mingw``
+         to install it.
+
+   #) Set up a Windows environment:
+
+      I) Download the :kbd:`Visual Studio 2019` tool from `<https://visualstudio.microsoft.com/downloads/>`_,
+         and choose the two options in the below screenshots to install "Microsoft Visual C++ Redistributable
+         for Visual Studio 2015, 2017 and 2019 (x86 or X64)" in WaaG:
+
+         .. figure:: images/Microsoft-Visual-C-install-option-1.png
+
+         .. figure:: images/Microsoft-Visual-C-install-option-2.png
+
+      #) In WaaG, use the :kbd:`WIN + R` shortcut key, input "shell:startup", click :kbd:`OK`
+         and then copy ``life_mngr_win.exe`` application into this directory.
+
+         .. figure:: images/run-shell-startup.png
+
+         .. figure:: images/launch-startup.png
+
+   #) Restart the WaaG VM. The COM2 window will automatically open after reboots.
+
+         .. figure:: images/open-com-success.png
+
+#. If the Service VM is being shut down (transitioning to the S5 state), it can call
+   ``acrnctl stop vm-name`` to shut down the User VMs.  A script is offered as reference
+   to be run as ``root``:
+
+   .. note:: S5 state is not automatically triggered by a Service VM shutdown, this needs
+      to be run before powering off the Service VM.
+
+   .. code-block:: bash
```

Comment:
I'm not fond of putting a script in the documentation where you're expecting the developer to cut and past it from the documentation into a local file?  Scripts must have a license header and copyright.